stellar-auth/pkg/auth/authenticator.go

```diff
@@ -522,6 +523,32 @@ func (a *defaultAuthenticator) GetUsers(ctx context.Context, userIDs []string) (
 	return users, nil
 }
 
+func (a *defaultAuthenticator) GetUserByEmail(ctx context.Context, email string) (*User, error) {
+	const query = `
+		SELECT
+			id,
+			email
+		FROM
+			auth_users
+		WHERE
+			email = $1 AND is_active = true
+	`
+
+	var u authUser
+	err := a.dbConnectionPool.GetContext(ctx, &u, query, email)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrUserNotFound
+		}
+		return nil, fmt.Errorf("querying user email %s: %w", email, err)
+	}
+
+	return &User{
+		ID:    u.ID,
+		Email: email,
+	}, nil
+}
+
```

Comment:
🤔 💡   Why don't you fetch the roles as well? I mean, 100% of the times where you fetch the user by email, you will need to look at the user roles as well, right? Aren;'t the roles on this very same table?

Reply:
TBH, `GetUser` could be returning these roles as well, but that doesn't need to be changed in this PR.